openshift_pool/openshift/stack.py

```diff
@@ -267,8 +281,37 @@ def hosts_data(self):
 
     def get_connection_statuses(self):
         """Return a LUT that contains each node and its connectivity by the domain"""
+
         out = {}
+
+        def test_ping(hostname):
+            self.log.debug(f"testing ping connectivity to host {hostname}")
+            result = subprocess.run(['ping', '-c', '1', hostname], stdout = subprocess.PIPE)
+            self.log.debug(f"[STDOUT]: {result.stdout}")
+            self.log.debug(f"[STDERR]: {result.stderr}")
+            self.log.debug(f"[RC]: {result.returncode}")
+
+            return bool(not result.returncode)
+
+        def test_ssh(hostname, username, password):
+            try:
+                self.log.debug(f"testing ssh connectivity to host {hostname} (creds: {username}\{password})")
+                is_ssh = False
+                client = paramiko.SSHClient()
+                client.load_system_host_keys()
+                client.set_missing_host_key_policy(paramiko.MissingHostKeyPolicy())
+                client.connect(hostname,username=username,password=password)
+                is_ssh = client.get_transport().isAlive()
+            except Exception:
+                pass
+            finally:
+                client.close()
+                return is_ssh
+
         for hostname in self.hosts_data['host_names'].values():
-            out[hostname] = not sp.run(
-                ['ping', '-c', '1', hostname]).returncode
+            ssh_state = test_ssh(hostname, self.ssh_details["username"], self.ssh_details["password"])
+            ping_state = test_ping(hostname)
```

Comment:
If the ping fails, should we really try to ssh?

Reply:
Well, I also considered that I think I will remove the ping test since SSH will not work if no ping...